arch/risc-v/src/mpfs/mpfs_ihc.c

```diff
@@ -163,19 +163,19 @@ static int mpfs_rptun_register_callback(struct rptun_dev_s *dev,
 
 uint8_t unused_filler[0x80000] __attribute__((section(".filler_area")));
 
-static struct   rpmsg_endpoint        g_mpgs_echo_ping_ept;
-static struct   mpfs_queue_table      g_mpfs_virtqueue_table[VRINGS];
-static struct   mpfs_rptun_shmem_s    g_shmem;
-static struct   rpmsg_device         *g_mpfs_rpmsg_device;
-static struct   rpmsg_virtio_device  *g_mpfs_virtio_device;
+static struct rpmsg_endpoint        g_mpgs_echo_ping_ept;
+static struct mpfs_queue_table      g_mpfs_virtqueue_table[VRINGS];
+static struct mpfs_rptun_shmem_s    g_shmem;
+static struct rpmsg_device         *g_mpfs_rpmsg_device;
+static struct rpmsg_virtio_device  *g_mpfs_virtio_device;
 
-static sem_t   g_mpfs_ack_lock      = SEM_INITIALIZER(1);
-static sem_t   g_mpfs_rx_lock       = SEM_INITIALIZER(1);
-static struct  list_node g_dev_list = LIST_INITIAL_VALUE(g_dev_list);
+static sem_t  g_mpfs_ack_lock      = SEM_INITIALIZER(1);
+static sem_t  g_mpfs_rx_sig        = SEM_INITIALIZER(1);
```

Comment:
@eenurkka are you sure that `g_mpfs_rx_sig` should be inited to `SEM_INITIALIZER(1)` and not to `SEM_INITIALIZER(0)`?

Reply:
Same for `g_mpfs_ack_lock` that also seems to be used for signalling (maybe can be renamed to `g_mpfs_ack_sig`)

Reply:
Could be, I grepped *.c files and figured everybody else uses SEM_INITIALIZER(1), so I went with that. Will check.

Reply:
```
static struct ramlog_dev_s g_sysdev =
{
#ifndef CONFIG_RAMLOG_NONBLOCKING
  0,                             /* rl_nwaiters */
#endif
  CONFIG_RAMLOG_BUFSIZE,         /* rl_head */
  CONFIG_RAMLOG_BUFSIZE,         /* rl_tail */
  SEM_INITIALIZER(1),            /* rl_exclsem */
#ifndef CONFIG_RAMLOG_NONBLOCKING
  SEM_INITIALIZER(0),            /* rl_waitsem */
#endif
  CONFIG_RAMLOG_BUFSIZE,         /* rl_bufsize */
  g_sysbuffer                    /* rl_buffer */
};
#endif
```
so `rl_exclsem` is mutual exclusion access guard while `rl_waitsem` is a signalling semaphore

Reply:
Yes, it would look like the zero is better here

Reply:
Created a separate PR#6261